### PR TITLE
CM-178: addressed seo issues with page and park templates 

### PR DESCRIPTION
**Pull request by foley-lynn**
### Jira Ticket:
CM-178

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/CM-178

### Description:
This is an attempt to fix some of the more egregious SEO errors on the page templates. Mainly has to do with meta tags.

### CODE REVIEW

File: src/staging/src/components/park/advisory.js [hunk 1]
```diff
@@ -56,6 +56,7 @@ export default function Advisory({ advisories }) {
             src={alertIcon}
             aria-label="park access status"
             className="park-overview-icon"
+            alt="icon"
```
Comment (molund) [inline src/staging/src/components/park/advisory.js:59]: Decorative images usually don’t need alt text.
I believe `alt=""` is actually better for screen readers.

File: src/staging/src/templates/staticLanding1.js [hunk 2]
```diff
@@ -70,9 +70,9 @@ const LandingPage = ({ pageContext }) => {
   return (
     <>
       <Seo
-        title={meta.metaTitle}
-        description={meta.description}
-        keywords={meta.metaKeywords}
+        title={meta?.metaTitle || pageContext.page.Title}
```
Comment (molund) [inline src/staging/src/templates/staticLanding1.js:73]: I think that should be page?.Title based on nearby examples

File: src/staging/src/components/park/advisory.js (hunk 1 above)
Comment (foley-lynn) [reply src/staging/src/components/park/advisory.js:59]: Thanks for catching this. This is a MUI component, and if you put `alt=""` in it strips it to `alt`, which is why I went with the above. I later discovered that from a screen-reader and SEO POV `alt=""` and `alt` are equivalent. So I'll switch all of those.

File: src/staging/src/templates/staticLanding1.js (hunk 2 above)
Comment (foley-lynn) [reply src/staging/src/templates/staticLanding1.js:73]: Thanks. page?.title wasn't working so I dug this up. I missed the capital 'T'.

**Review by molund — APPROVED** [review]

**Review by ayumi-oxd — COMMENTED** [review]
I have some q's 🙂
- Do we need to have `seo` component for `site` page as the same as `park` page?
- Do we need to pull `seo` component out from the content list at `page` collection? To make it consistent with `protectedArea` and `parkSubPage` collections.

Other than that it looks good to me 👏
- I was able to see `seo` component on `protectedArea`
- I was able to build this PR with DEV, TEST, and PROD Strapi environment on my local

File: src/staging/gatsby-node.js [hunk 3]
```diff
@@ -227,6 +228,7 @@ exports.createSchemaCustomization = ({ actions }) => {
     metaKeywords: String
     metaTitle: String
     metaDescription: String
+    canonical: String
```
[the remark this answers is not in the document]
Comment (foley-lynn) [reply src/staging/gatsby-node.js:231]: Good catch, I was actually implementing in incorrectly. this has been removed.

File: src/staging/src/templates/parkSubPage.js [hunk 4]
```diff
@@ -80,7 +80,7 @@ export default function ParkSubPage({ data }) {
     <>
       <Seo
         title={seo?.metaTitle ?? park.protectedAreaName + ": " + page.title}
-        description={seo?.description}
+        description={seo?.metaDescription}
         keywords={seo?.metaKeywords}
       />
```
[the remark this answers is not in the document]
Comment (foley-lynn) [reply src/staging/src/templates/parkSubPage.js:85]: No, I was implementing the canonical links incorrectly. With the plugin, they're on all pages automatically.

**Comment by foley-lynn** [comment]
> * Do we need to have `seo` component for `site` page as the same as `park` page?

Yes, I've now added that.

> * Do we need to pull `seo` component out from the content list at `page` collection? To make it consistent with `protectedArea` and `parkSubPage` collections.

Probably, but it's messy. The whole way the pages are set up might just be better to not touch.

File: src/cms/api/protected-area/models/protected-area.settings.json [hunk 5]
```diff
@@ -181,6 +181,11 @@
     "parkSubPages": {
       "via": "protectedArea",
       "collection": "park-sub-page"
+    },
+    "seo": {
+      "type": "component",
+      "repeatable": false,
+      "component": "parks.seo"
     }
```
Comment (ayumi-oxd) [inline src/cms/api/protected-area/models/protected-area.settings.json:189]: I think you'd need to add the same setting to `site.setting.json` to use `seo` component in `site.js`

File: src/staging/gatsby-node.js [hunk 6]
```diff
@@ -104,6 +104,7 @@ exports.createSchemaCustomization = ({ actions }) => {
     parkFacilities: [StrapiParkFacilities]
     parkOperation: StrapiParkOperation
     parkOperationSubAreas: [StrapiParkOperationSubAreas]
+    seo: StrapiSeoComponent
   }
```
Comment (ayumi-oxd) [inline src/staging/gatsby-node.js:108]: I think you'd need to define type for `seo` at `site` collection. Adding `seo: StrapiSeoComponent` to `type StrapiSites implements Node` should work.

**Review by ayumi-oxd — COMMENTED** [review]
Added a few comments, but other than that it looks great 👏

File: src/cms/api/protected-area/models/protected-area.settings.json (hunk 5 above)
Comment (foley-lynn) [reply src/cms/api/protected-area/models/protected-area.settings.json:189]: I'm using the component in the front-end, but not using any of the fields in site pages. For now it's just using the park and site name for the title, and truncating the description for the meta description.

We should integrated the SEO component in eventually, but not for MVP.

File: src/staging/gatsby-node.js (hunk 6 above)
Comment (foley-lynn) [reply src/staging/gatsby-node.js:108]: As above, for sites we're just using the component to populate the title and the description based on existing fields. So I think it's ok to not register it here. I might be wrong though, so if you know better, let me know.

**Review by ayumi-oxd — APPROVED** [review]
Sorry I didn't notice `site.js` doesn't use data from `Seo`. Works great as is 👍

**Pull request merged**